test/ngAria/ariaSpec.js

```diff
@@ -104,18 +104,48 @@ describe('$aria', function() {
   describe('aria-checked', function() {
     beforeEach(injectScopeAndCompiler);
 
-    it('should attach itself to input type="checkbox"', function() {
+    it('should not attach itself to native input type="checkbox"', function() {
       compileElement('<input type="checkbox" ng-model="val">');
 
       scope.$apply('val = true');
+      expect(element.attr('aria-checked')).toBeUndefined();
+
+      scope.$apply('val = false');
+      expect(element.attr('aria-checked')).toBeUndefined();
+    });
+
+    it('should attach itself to custom checkbox', function() {
+      compileElement('<div role="checkbox" ng-model="val">');
+
+      scope.$apply('val = true');
       expect(element.attr('aria-checked')).toBe('true');
 
       scope.$apply('val = false');
       expect(element.attr('aria-checked')).toBe('false');
     });
 
-    it('should handle checkbox with string model values using ng(True|False)Value', function() {
-      var element = $compile('<input type="checkbox" ng-model="val" ng-true-value="\'yes\'" ' +
+    it('should not handle native checkbox with ngChecked', function() {
+      var element = $compile('<input type="checkbox" ng-checked="val">')(scope);
+
+      scope.$apply('val = true');
+      expect(element.eq(0).attr('aria-checked')).toBeUndefined();
+
+      scope.$apply('val = false');
+      expect(element.eq(0).attr('aria-checked')).toBeUndefined();
+    });
+
+    it('should handle custom checkbox with ngChecked', function() {
+      var element = $compile('<div role="checkbox" ng-checked="val">')(scope);
+
+      scope.$apply('val = true');
+      expect(element.eq(0).attr('aria-checked')).toBe('true');
+
+      scope.$apply('val = false');
+      expect(element.eq(0).attr('aria-checked')).toBe('false');
+    });
+
+    it('should handle custom checkbox with string model values using ng(True|False)Value', function() {
+      var element = $compile('<div role="checkbox" ng-model="val" ng-true-value="\'yes\'" ' +
```

Comment:
If you wnat to be sure that the behavior is tested correctly, you should use a falsy value for `ngTrueValue` and a truthy value for `ngFalseValue` (e.g. `ng-true-value="0" ng-false-value="1"`).